portfolio/src/main/webapp/paragraph.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Adds a headig with a paragraph to each picture.
+ */
+let files = getImages("/images");
+let descriptions = ["This picture was taken accidentally by my friend two years ago. However it becomes one of my favorite photo since at that time I was still quite slim.",
+                    "Me & Welcome Packge with Google!",
+                    "Welcome package from Google!",
+                    "My favorite ID photo, since I was truly slim at that time."];
+let size = files.length;
+for (let i = 0; i < size; i++) {
+    let tag = document.createElement("a");
+    let source = document.createAttribute("href");
+    source.value = files[i]; 
+    tag.setAttributeNode(source);
+    tag.innerHTML = "<h3>P" + (i + 1) + "</h3>";
+
+    let description = document.createElement("p");
+    let text = document.createTextNode(descriptions[i]);
+    description.appendChild(text);
+
+    let element = document.getElementById("content");
+    element.appendChild(tag);
+    element.appendChild(description);
+}
+
+/**
+ * Gets the images to be displayed.
+ * @param {string} directory  directory to the images.
+ * @return {string array} an array of file name string.
+ */
+function getImages(directory) {
```

Comment:
This method is not making use of the directory input?

Reply:
I design the function to take in the directory name and later to figure out a way to get the file names in the folder, but so far the file name is hard coded.

Reply:
Fixed. Make the path name as string concatenation